File: brain-move/games/diario_missioni.py
import time
from games.base_game import BaseGame


class DiarioMissioni(BaseGame):
    """Gioco 6: Diario sessione - feedback finale e memorizzazione leggera."""

    QUESTIONS = [
        "Quale gioco ti e piaciuto di piu oggi?",
        "Ti senti calmo, normale o stanco?",
        "Vuoi rifare lo stesso percorso domani?",
    ]

    CAREGIVER_NOTES = [
        "attenzione_buona",
        "affaticamento",
        "umore_positivo",
        "bisogno_di_pausa",
        "difficolta_memoria",
        "difficolta_movimento",
    ]

    def __init__(self, difficulty_engine, scoring_engine):
        super().__init__(difficulty_engine, scoring_engine)
        self.name = "diario_missioni"
        self.display_name = "Diario delle Missioni"
        self.current_question = 0
        self.answers = []
        self.caregiver_notes = []
        self.phase = "questions"
        self.timer = 0

    def start(self):
        super().start()
        self.current_question = 0
        self.answers = []
        self.caregiver_notes = []
        self.phase = "questions"
        self.timer = time.time()
        self.feedback_message = self.QUESTIONS[0]

    def answer_question(self, answer):
        if self.phase != "questions":
            return
        self.answers.append(answer)
        self.current_question += 1
        if self.current_question >= len(self.QUESTIONS):
            self.phase = "caregiver"
            self.feedback_message = "Caregiver: seleziona le osservazioni"
        else:
            self.feedback_message = self.QUESTIONS[self.current_question]

    def add_caregiver_note(self, note):
        if note in self.CAREGIVER_NOTES:
            self.caregiver_notes.append(note)

    def complete(self):
        self.result = {
            "completed": True,
            "answers": dict(zip(self.QUESTIONS, self.answers)),
            "caregiver_notes": self.caregiver_notes,
        }
        self.scoring.add_score(self.name, 10)
        self.stop()
        self.feedback_message = "Sessione completata, ottimo lavoro!"

    def get_current_question(self):
        if self.phase == "questions" and self.current_question < len(self.QUESTIONS):
            return self.QUESTIONS[self.current_question]
        return None

    def get_caregiver_options(self):
        return self.CAREGIVER_NOTES

    def get_progress(self):
        if self.phase == "questions":
            return self.current_question / max(len(self.QUESTIONS), 1) * 0.7
        return 0.7 + 0.3 * min(len(self.caregiver_notes) / 3, 1.0)

File: brain-move-web/src/games/diario-missioni.ts
import { BaseGame } from './base-game'
import { Difficulty } from '../engine/difficulty'
import { Scoring } from '../engine/scoring'

const QUESTIONS = [
  "Quale gioco ti è piaciuto di più oggi?",
  "Ti senti calmo, normale o stanco?",
  "Vuoi rifare lo stesso percorso domani?",
]

const CAREGIVER_NOTES = [
  "attenzione_buona", "affaticamento", "umore_positivo",
  "bisogno_di_pausa", "difficolta_memoria", "difficolta_movimento",
]

const CAREGIVER_LABELS: Record<string, string> = {
  attenzione_buona: "Attenzione buona",
  affaticamento: "Affaticamento",
  umore_positivo: "Umore positivo",
  bisogno_di_pausa: "Bisogno di pausa",
  difficolta_memoria: "Difficoltà memoria",
  difficolta_movimento: "Difficoltà movimento",
}

export class DiarioMissioni extends BaseGame {
  currentQuestion = 0
  answers: string[] = []
  caregiverNotes: string[] = []
  phase: "questions" | "caregiver" = "questions"

  constructor(difficulty: Difficulty, scoring: Scoring) {
    super(difficulty, scoring)
    this.name = "diario_missioni"
    this.displayName = "Diario delle Missioni"
  }

  start(): void {
    super.start()
    this.currentQuestion = 0
    this.answers = []
    this.caregiverNotes = []
    this.phase = "questions"
    this.feedbackMessage = QUESTIONS[0]
  }

  update(_poseData?: any, _dt?: number): void {}

  answerQuestion(answer: string): void {
    if (this.phase !== "questions") return
    this.answers.push(answer)
    this.currentQuestion++
    if (this.currentQuestion >= QUESTIONS.length) {
      this.phase = "caregiver"
      this.feedbackMessage = "Caregiver: seleziona le osservazioni"
    } else {
      this.feedbackMessage = QUESTIONS[this.currentQuestion]
    }
  }

  addCaregiverNote(note: string): void {
    if (CAREGIVER_NOTES.includes(note) && !this.caregiverNotes.includes(note))
      this.caregiverNotes.push(note)
  }

  removeCaregiverNote(note: string): void {
    this.caregiverNotes = this.caregiverNotes.filter(n => n !== note)
  }

  complete(): void {
    this.result = { completed: true, accuracy: 1, answers: this.answers, caregiverNotes: this.caregiverNotes }
    this.scoring.addScore(this.name, 10)
    this.stop()
    this.feedbackMessage = "Sessione completata, ottimo lavoro!"
  }

  getCurrentQuestion(): string | null {
    return this.phase === "questions" && this.currentQuestion < QUESTIONS.length ? QUESTIONS[this.currentQuestion] : null
  }

  getCaregiverOptions(): string[] { return CAREGIVER_NOTES }
  getCaregiverLabel(note: string): string { return CAREGIVER_LABELS[note] ?? note }

  getProgress(): number {
    if (this.phase === "questions") return this.currentQuestion / Math.max(QUESTIONS.length, 1) * 0.7
    return 0.7 + 0.3 * Math.min(this.caregiverNotes.length / 3, 1)
  }
}
